airflow/bin/cli.py

```diff
@@ -336,6 +336,83 @@ def list_tasks(args, dag=None):
         print("\n".join(sorted(tasks)))
 
 
+def get_start_date(dag):
```

Comment:
It's a good idea to centralize this logic, since it's also used by the scheduler. I think a private method like `dag._get_start_date()` would be best since we already have a `start_date` attribute that would be returned by the method unless `start_date is None`.